pkg/provider-local/controller/worker/actuator.go

```diff
@@ -32,28 +32,47 @@ import (
 	"github.com/go-logr/logr"
 	"k8s.io/client-go/kubernetes"
 	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
 )
 
 type delegateFactory struct {
 	logger logr.Logger
 	common.RESTConfigContext
 }
 
+type actuator struct {
+	worker.Actuator
+}
+
 // NewActuator creates a new Actuator that updates the status of the handled WorkerPoolConfigs.
 func NewActuator() worker.Actuator {
 	delegateFactory := &delegateFactory{
 		logger: log.Log.WithName("worker-actuator"),
 	}
 
-	return genericactuator.NewActuator(
-		log.Log.WithName("local-worker-actuator"),
-		delegateFactory,
-		local.MachineControllerManagerName,
-		mcmChart,
-		mcmShootChart,
-		imagevector.ImageVector(),
-		extensionscontroller.ChartRendererFactoryFunc(util.NewChartRendererForShoot),
-	)
+	return &actuator{
+		genericactuator.NewActuator(
+			log.Log.WithName("local-worker-actuator"),
+			delegateFactory,
+			local.MachineControllerManagerName,
+			mcmChart,
+			mcmShootChart,
+			imagevector.ImageVector(),
+			extensionscontroller.ChartRendererFactoryFunc(util.NewChartRendererForShoot),
+		),
+	}
+}
+
+func (a *actuator) InjectFunc(f inject.Func) error {
+	return f(a.Actuator)
+}
+
+func (a *actuator) Migrate(ctx context.Context, worker *extensionsv1alpha1.Worker, cluster *extensionscontroller.Cluster) error {
+	return a.Actuator.Delete(ctx, worker, cluster)
```

Comment:
The delete call here will make sure that the `Node` object that is registered for the shoot cluster is also removed. Otherwise, when the new `Machine` and worker node get created there will be two `Node` objects registered in the cluster. The old one will be stuck in a `NotReady` state as it does not have any backing `Pod` in the shoot's control plane namespace, pods that were scheduled on it will also be stuck in `Terminating` or `Unknown` status and this can lead to errors during the restore phase. 
I've added comments in both the Restore and Migrate methods to try to explain this.